tests/BasicSearchTests.FunctionalTests.Core/StartupFunctionalTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net;
+using System.Threading.Tasks;
+using Xunit;
+using System;
+using System.Net.Http;
+using BasicSearchTests.FunctionalTests.Core.Models;
+using Newtonsoft.Json;
+
+namespace BasicSearchTests.FunctionalTests.Core
+{
+    public class StartupFunctionalTests
+    {
+        private HttpClient _client;
+        private static string RegistrationBaseUrl = "RegistrationsBaseUrl";
+
+        public StartupFunctionalTests()
+        {
+            // Arrange
+            _client = new HttpClient { BaseAddress = new Uri(EnvironmentSettings.SearchServiceBaseUrl) };
+        }
+
+        [Fact]
+        public async Task Ready()
+        {
+            // Act
+            var response = await _client.GetAsync("/");
+            var content = await response.Content.ReadAsStringAsync();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(content, "READY");
+        }
+
+        [Fact]
+        public async Task InvalidEndpoint()
+        {
+            // Act
+            var response = await _client.GetAsync("/invalid");
+            var content = await response.Content.ReadAsStringAsync();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+            Assert.Equal(content, "UNRECOGNIZED");
+        }
+
+        [Fact]
+        public async Task IndexIsFresh()
+        {
+            var httpClient = new HttpClient();
```

Comment:
>HttpClient [](start = 33, length = 10)

something that concerns me about this code in general: requests are not retried. Since we run this code during deployment, a failure of a request will cause the deployment to fail. NuGet.Protocol package contains a class called HttpRetryHandler. Using it will allow you to easily add retries to your code.